homeassistant/components/vegehub/sensor.py

```diff
@@ -0,0 +1,210 @@
+"""Sensor configuration for VegeHub integration."""
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfElectricPotential, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CHAN_TYPE_BATTERY,
+    CHAN_TYPE_SENSOR,
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+    OPTION_DATA_TYPE_CHOICES,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Vegetronix sensors from a config entry."""
+    sensors = []
+    mac_address = str(config_entry.data.get("mac_address"))
+    ip_addr = str(config_entry.data.get("ip_addr"))
+    num_sensors = int(config_entry.data.get("hub", {}).get("num_channels") or 0)
+    is_ac = int(config_entry.data.get("hub", {}).get("is_ac") or 0)
```

Comment:
same here, when would those not be set?

Reply:
Fixed in commit be1232c7ecd511885e79d69133aea2ed73d159db